src/main/java/org/broadinstitute/hellbender/engine/dataflow/datasources/ShardedReadsDatasource.java

```diff
@@ -0,0 +1,382 @@
+                private transient Stopwatch loading;
+                private transient Stopwatch choosing;
+                private transient Stopwatch shipping;
+                private long readsEmitted = 0;
+                Date started;
+
+                @Override
+                public void startBundle(Context c) throws Exception {
+                    super.startBundle(c);
+                    if (cloudStorageUrl) {
+                        storageClient = GCSOptions.Methods.createStorageClient(c.getPipelineOptions().as(GCSOptions.class), auth);
+                    }
+                    started = new Date();
+                    readsEmitted = 0;
+                    querying = Stopwatch.createUnstarted();
+                    loading = Stopwatch.createUnstarted();
+                    choosing = Stopwatch.createUnstarted();
+                    shipping = Stopwatch.createUnstarted();
+                }
+
+                @Override
+                public void finishBundle(Context c) throws Exception {
+                    super.finishBundle(c);
+                    final Instant instant = new Date().toInstant();
+                    long ms = Duration.between(started.toInstant(), instant).toMillis();
+                    String q = querying.elapsed(TimeUnit.MILLISECONDS) + " ms";
+                    String l = loading.elapsed(TimeUnit.MILLISECONDS) + " ms";
+                    String ch = choosing.elapsed(TimeUnit.MILLISECONDS) + " ms";
+                    String s = shipping.elapsed(TimeUnit.MILLISECONDS) + " ms";
+                    bunny.stepEnd("Bundle sent " + readsEmitted + " reads in " + ms + " ms. Breakdown: querying " + q + " loading " + l + " choosing " + ch + " shipping " + s);
+                }
+
+                @Override
+                public void processElement(ProcessContext c) throws Exception {
+                    int rejected = 0;
+                    int accepted = 0;
+                    Stopwatch processing = Stopwatch.createStarted();
+
+                    Stopwatch kving = Stopwatch.createUnstarted();
+                    Stopwatch outputting = Stopwatch.createUnstarted();
+
+                    Stopwatch readingInput = Stopwatch.createStarted();
+                    SimpleInterval i = c.element();
+                    readingInput.stop();
+                    Stopwatch init = Stopwatch.createStarted();
+                    ArrayList<SimpleInterval> is = new ArrayList<>();
+                    is.add(i);
+
+                    Stopwatch cutShards = Stopwatch.createStarted();
+                    List<SimpleInterval> shards = cutToShards(is, outputShardSize);
+                    cutShards.stop();
+
+                    int currentShardIndex = 0;
+                    SimpleInterval currentShard = shards.get(currentShardIndex);
+                    List<GATKRead> ret = new ArrayList<GATKRead>();
+                    init.stop();
+
+                    if (!cloudStorageUrl && !hadoopUrl) {
```

Comment:
When loading from a local file, it looks like you fail to set the header in the underlying SAMRecords to null. If the method claims to strip the header from the reads, it should do so in all cases.